test/versioned/langchain/runnables.tap.js

```diff
@@ -95,11 +94,35 @@ tap.test('Langchain instrumentation - runnable sequence', (t) => {
     })
   })
 
+  t.test('should support custom attributes on the LLM events', (t) => {
+    const { agent, prompt, outputParser, model } = t.context
+    const api = helper.getAgentApi()
+    helper.runInTransaction(agent, async (tx) => {
+      api.withLlmCustomAttributes({ 'llm.contextAttribute': 'someValue' }, async () => {
+        const input = { topic: 'scientist' }
+        const options = { metadata: { key: 'value', hello: 'world' }, tags: ['tag1', 'tag2'] }
+
+        const chain = prompt.pipe(model).pipe(outputParser)
+        await chain.invoke(input, options)
+        const events = agent.customEventAggregator.events.toArray()
+        const responses = events.filter((event) => {
+          const [, chainEvent] = event
+          return chainEvent.virtual_llm === true && chainEvent.is_response === true
+        })
+        const [[, message]] = responses
+        t.ok(message['llm.contextAttribute'])
```

Comment:
`t.ok` is only asserting that the field exists with a truthy value (i.e. not `null` or `undefined`).

```suggestion
        t.equal(message['llm.contextAttribute'], 'someValue')
```

^ asserts that the field exists _and_ has the expected value.

Reply:
We get 2 sets of similar events from langchain and openai instrumentations.  Langchain instrumentation would record its events with a proper llm context but having a flag of virtual_llm=true but openai would record events under a different context and with virtual_llm flag off.

Reply:
Yeah. It was not available in the current context when recordEvent was called for open AI. But I just committed a solution to use an isolated context manager. 

Let me know what you think

Reply:
I don't get how creating a different context maanger has fixed this.  I'll look later today but I don't think we should be creating a new context manager, there is something else happening here

Reply:
I suspect we run all Open AI calls in a different context derived from Langchain context. I could try a solution 2 mentioned in the description to see if the llm context is left on one of the parent contexts.

Reply:
I filed this [ticket](https://github.com/newrelic/node-newrelic/issues/2495). we can update this code after it lands to use the agent context manager but a few things have to be fixed before we do that